Sources/Fuzzilli/FuzzIL/Operations.swift

```diff
@@ -622,6 +622,31 @@ class Reassign: Operation {
     }
 }
 
+/// Destructs an array into n output variables
+class DestructArray: Operation {
+    let indices: [Int]
+    let hasRestElement: Bool
+    
+    init(indices: [Int], hasRestElement: Bool) {
+        self.hasRestElement = hasRestElement
```

Comment:
`indices` must be sorted here, correct? Maybe add an `assert` to verify that? I guess it must also not have duplicates, although that might be a bit harder to assert (maybe convert to a `Set`, then compare against that?), so if it's more than one short-ish line, it's probably fine to just add a comment for that property